test/e2e/rule_test.go

```diff
@@ -317,11 +345,16 @@ func TestRule(t *testing.T) {
 		rulegroupCorrectData(t, ctx, r.HTTPEndpoint())
 	})
 
-	t.Run("reload works", func(t *testing.T) {
-		// Add a new rule via /-/reload.
-		// TODO(GiedriusS): add a test for reloading via SIGHUP. Need to extend e2e framework to expose PIDs.
+	t.Run("signal reload works", func(t *testing.T) {
+		// Add a new rule via sending sighup
+		createRuleFile(t, fmt.Sprintf("%s/newrule.yaml", rulesPath), testAlertRuleAddedLater)
+		reloadRulesSignal(t, r)
+		checkReloadSuccessful(t, ctx, r.HTTPEndpoint())
+	})
 
-		createRuleFile(t, fmt.Sprintf("%s/newrule.yaml", rulesPath), testAlertRuleAddedLaterWebHandler)
+	t.Run("http reload works", func(t *testing.T) {
+		// Add a new rule via /-/reload.
+		createRuleFile(t, fmt.Sprintf("%s/newrule.yaml", rulesPath), testAlertRuleAddedLater)
```

Comment:
The test later on here creates a file with the same name & content. Maybe we could have one `testAlertRuleAddedLater` with 3 rule groups & another one with 4 rule groups, let's say? And then use the one with 4 rule groups here. After refactoring `checkReloadSuccessful` to accept the expected rule group count, we could use that function here as well.

Reply:
Could you check the last commit if understood correctly what you suggested please?